### DISCUSSION TITLE
Reverse Specifics

### DISCUSSION THREAD

**Discussion by GoldenStack** [TodePond/GulfOfMexico#418] [Q&A] [answered]
```java
if false {
   print Hey !
}
reverse!
```
Does this print `Hey`? What about similar code involving `when`?

**Comment by TodePond** [comment, marked as answer]
Imagine it like this:

```java
if false {
   print Hey !
}
}
   print Hey !
if false {
```

Line 4 and 6 get interpreted as implicit strings, so yes - it does print Hey

**Reply by GoldenStack** [reply]
Even with functions? So if I do this:
```java
function a b {
   reverse!
   return 1;
}
```
This would completely escape out of the function and interpret `function a b {` as an implicit string (as opposed to returning `undefined`)?
That's actually amazing. I'm going to have a lot of fun parsing this

**Comment by TodePond** [comment]
Not sure what you mean about `when` though. it works quite differently

**Reply by GoldenStack** [reply]
Basically I was thinking if this would print `You lose`
```java
when false {
   print You lose!
}
reverse!
```
Although your previous message suggests it would apply the same way for this too.

(I was assuming that it would parse statements backwards, instead of parsing individual lines backwards; thanks for clearing it up)
